galley/pkg/config/analysis/diag/message.go

```diff
@@ -70,7 +74,13 @@ func (m *Message) Unstructured(includeOrigin bool) map[string]interface{} {
 	if includeOrigin && m.Resource != nil {
 		result["origin"] = m.Resource.Origin.FriendlyName()
 		if m.Resource.Origin.Reference() != nil {
-			result["reference"] = m.Resource.Origin.Reference().String()
+			var loc string
+			if len(m.line) != 0 && m.line[0] != 0 {
+				loc = " " + m.Resource.Origin.Reference().String(true) + fmt.Sprintf("%d", m.line[0])
+			} else {
+				loc = " " + m.Resource.Origin.Reference().String(false)
+			}
```

Comment:
I believe the Reference already knows whether it has a line number or not, so you should be able to put the string formatting logic into String() itself and not need a different function.